src/app/api/user/page_views/route.ts

```diff
@@ -1,118 +1,192 @@
+    if (!session) {
+      return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
+    }
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
+    if (!user) {
+      return NextResponse.json({ error: "User not found" }, { status: 404 });
+    }
+    const url = new URL(req.url);
+    const queryUrl = Object.fromEntries(url.searchParams);
+    const parsedResult = getSchema.safeParse({
+      page: queryUrl.page ? Number(queryUrl.page) : undefined,
+      limit: queryUrl.limit ? Number(queryUrl.limit) : undefined,
+      distinct: queryUrl.distinct ? Boolean(queryUrl.distinct) : undefined,
+    });
+    if (!parsedResult.success) {
+      return NextResponse.json(
+        {
+          error: parsedResult.error.errors[0].message,
+          errors: parsedResult.error.errors.map((err) => err.message),
+        },
+        { status: 404 },
+      );
+    }
+    const { page, limit, distinct } = parsedResult.data;
+    const countPageViewAnchor = await prisma.pageView.findMany({
       orderBy: {
-        _count: {
-          storyId: "desc",
+        createdAt: "desc",
+      },
+      select: {
+        storyId: true,
+        createdAt: true,
+        story: {
+          select: {
+            title: true,
+            storyContributions: {
+              select: {
+                contributorId: true,
+              },
+            },
+            thumbnailUrl: true,
+          },
         },
       },
-      take: 10,
+      skip: page * limit,
+      take: limit,
+      distinct: distinct ? ["storyId"] : undefined,
       where: {
-        userId: user_id,
+        userId: user.id,
       },
     });
+    //this will help get distinct story id from user reading history
+    const storyIdSet = new Set<string>();
+    countPageViewAnchor.forEach((pageView) => storyIdSet.add(pageView.storyId));
+    //get bookmark and brain
+    const mapStoryBookmarkedPromise = getMarkedStories(
+      user.id,
+      Array.from(storyIdSet),
+      "BOOK_MARK",
+    );
+    //get brained
+    const mapStoryBrainedPromise = getMarkedStories(
+      user.id,
+      Array.from(storyIdSet),
+      "BRAIN",
+    );
 
-    const storyIds = result.map((item) => item.storyId);
-
-    const stories = await prisma.story.findMany({
+    const countPageViewsPromise = prisma.pageView.findMany({
       where: {
-        id: {
-          in: storyIds,
-        },
+        userId: user.id,
       },
+      distinct: distinct ? ["storyId"] : undefined,
+      select: { id: true },
+    });
+    const [countPageViews, mapStoryBookmarked, mapStoryBrained] =
+      await Promise.all([
+        countPageViewsPromise,
+        mapStoryBookmarkedPromise,
+        mapStoryBrainedPromise,
+      ]);
+    return NextResponse.json({
+      total: countPageViews.length,
+      stories_history: countPageViewAnchor.map(
+        ({ storyId, createdAt, story }) => ({
+          story_id: storyId,
```

Comment:
if you're including the full stories history, would it make more sense to include title and possibly summary?  As well as possibly the contributor names?